pkg/graveler/staging/manager.go

```diff
@@ -67,16 +69,15 @@ func (m *Manager) Set(ctx context.Context, st graveler.StagingToken, key gravele
 	pb := graveler.ProtoFromStagedEntry(key, value)
 	stPartition := graveler.StagingTokenPartition(st)
 	if requireExists {
-		return m.store.SetMsgIf(ctx, stPartition, key, pb, kv.PrecondConditionalExists)
+		return kv.SetMsgIf(ctx, m.kvStore, stPartition, key, pb, kv.PrecondConditionalExists)
```

Comment:
Manager has both stores no? Why change the signature?

Reply:
The rate-limited store uses Store interface not a message store interface.
Didn't want to apply a complete makeover of using the msg store in this PR. I think we can remove the wrapper and just keep the functions as it extend the store interface and not wrap it.

Reply:
Maybe it's a silly question, but can't StoreLimiter accept a StoreMessage and implement the StoreMessage interface as well?